src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        Node<T> nodeToInsert = new Node<>(null, value, null);
+
+        if (index == 0) {
+            nodeToInsert.next = head;
+            head.prev = nodeToInsert;
+            head = nodeToInsert;
+            size++;
+            return;
+        }
+
+        Node<T> current = getNodeByIndex(index);
+
+        nodeToInsert.prev = current.prev;
+        nodeToInsert.next = current;
+        current.prev.next = nodeToInsert;
+        current.prev = nodeToInsert;
+        size++;
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (T element : list) {
+            add(element);
+        }
+
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeByIndex = getNodeByIndex(index);
+        return nodeByIndex.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> nodeByIndex = getNodeByIndex(index);
+        T previousValue = nodeByIndex.element;
+        nodeByIndex.element = value;
+        return previousValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+
+        Node<T> current;
+        T element;
+
+        if (index == 0) {
+            element = head.element;
+            head = head.next;
+        } else {
+            current = getNodeByIndex(index);
+            element = current.element;
+            current.prev.next = current.next;
+
+            if (current.next != null) {
+                current.next.prev = current.prev;
+            }
+        }
+        size--;
```

Comment:
Due to Common mistakes you should to create new method unlink and place there this part of your code. But the line with initializing of current  and element you can leave in remove method.
 `current = getNodeByIndex(index); element = current.element;`

Reply:
and empty lines